h/services/bulk_api/group.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+from typing import List
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, Group
+from h.services.bulk_api.core import date_match
+
+
+@dataclass
+class BulkGroup:
+    authority_provided_id: str
+
+
+class BulkGroupService:
+    """A service for retrieving groups in bulk."""
+
+    def __init__(self, db: Session):
+        self._db = db
+
+    def group_search(
+        self, groups: List[str], annotations_created: dict
+    ) -> List[BulkGroup]:
+        query = (
+            sa.select([Group.authority_provided_id])
+            .join(Annotation, Group.pubid == Annotation.groupid)
+            .group_by(Group.authority_provided_id)
+            .where(
+                date_match(Annotation.created, annotations_created),
+                Group.authority_provided_id.in_(groups),
+            )
+        )
+        results = self._db.scalars(query)
+        return [BulkGroup(authority_provided_id=row) for row in results.all()]
```

Comment:
Does this need to be limited to only groups belonging to the LMS authority? (Or only belonging to the authority that the current request is authenticated as)

Reply:
No sure. I don't it adds anything security wise